cps/main.c

```diff
@@ -59,14 +59,14 @@ cleanup_cps(void)
 
 	rte_timer_stop(&cps_conf.scan_timer);
 	destroy_mailbox(&cps_conf.mailbox);
-	rm_kni();
 	/*
 	 * rd_event_sock_close() can be called even when the netlink
 	 * socket is not open, and rte_kni_release() can be passed NULL.
 	 */
 	rd_event_sock_close(&cps_conf);
 	rte_kni_release(cps_conf.back_kni);
 	rte_kni_release(cps_conf.front_kni);
+	rm_kni();
```

Comment:
While this patch fixes the bug, the real issue is that `cleanup_cps()` is not doing the releases in the reverse order. Please inspect `run_cps()`, `cps_stage1()`, `cps_stage2()`, and related functions to identify the proper reverse order.

Reply:
In ``run_cps()``, the kni module, mailbox and timer will be initialized before the functions ``cps_stage1()``, ``cps_stage2()`` and ``cps_stage3()`` that are really called.

In ``cps_stage1()``, the mempool and KNIs are created. In ``cps_stage2()``, ``rd_event_sock_open()`` is called to open a Netlink socket.

In the reverse order, we should (1) close the Netlink socket; (2) release KNIs; (3) release the mempool; (4) stop the timer, release mailbox and kni kernel module.